actionmailer/lib/action_mailer/new_delivery_job.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require "active_job"
+
+module ActionMailer
+  # The <tt>ActionMailer::NewDeliveryJob</tt> class is used when you
+  # want to send emails outside of the request-response cycle. It supports
+  # sending either parameterized or normal mail.
+  #
+  # Exceptions are rescued and handled by the mailer class.
+  class NewDeliveryJob < ActiveJob::Base # :nodoc:
+    queue_as { ActionMailer::Base.deliver_later_queue_name }
+
+    rescue_from StandardError, with: :handle_exception_with_mailer_class
+
+    def perform(mailer, mail_method, delivery_method, params:, args:) #:nodoc:
+      mailer_class = params ? mailer.constantize.with(params) : mailer.constantize
+      mailer_class.public_send(mail_method, *args).send(delivery_method)
+    end
```

Comment:
Sorry missed this on the first round of review - I think `params` should be `nil` by default right? Since parameter jobs are a special kind of job?